test/test-sortedarray.c

```diff
@@ -0,0 +1,241 @@
+/*
+
+Copyright (c) 2016, Stefan Cloudt
+
+Permission to use, copy, modify, and/or distribute this software
+for any purpose with or without fee is hereby granted, provided
+that the above copyright notice and this permission notice appear
+in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+
+#include "alloc-testing.h"
+#include "framework.h"
+
+#include "sortedarray.h"
+
+#define TEST_SIZE 100
+
+int compare_float(void* v1, void* v2) 
+{
+    float* f1 = (float*) v1;
+    float* f2 = (float*) v2;
+
+    if (f1 == f2) {
+        return 0;
+    }
+
+    if (f1 == NULL || f2 == NULL) {
+        return (f1 == NULL) ? 1 : -1;
+    }
+
+    //discard fractional part, for testing purposes.
+    int i1 = (int) *f1;
+    int i2 = (int) *f2;
+
+    if (i1 == i2) {
+        return 0;
+    }
+
+    return (i1 > i2) ? 1 : -1;
+}
+
+int equal_float(void* v1, void* v2)
+{
+    float* f1 = (float*) v1;
+    float* f2 = (float*) v2;
+
+    if (f1 == f2) {
+        return 1;
+    }
+
+    if (f1 == NULL || f2 == NULL ) {
+        return 0;
+    }
+
+    return (*f1 == *f2) ? 1 : 0;
+}
+
+void check_sorted_prop(SortedArray *sortedarray)
+{
+    unsigned int i;
+    for (i = 1; i < sortedarray->array->length; i++) {
+        float *pf = *(sortedarray->array->data + i);
+        assert(compare_float(pf - 1, pf) <= 0);
+    }
+}
+
+void free_sorted_floats(SortedArray *sortedarray)
+{
+    unsigned int i;
+    for (i = 0; i < sortedarray->array->length; i++) {
+        float *pf = *(sortedarray->array->data + i);
+        free(pf);
+    }
+
+    sortedarray_free(sortedarray);
+}
+
+SortedArray *generate_sortedarray(void)
+{
+    //generate a sorted array of length TEST_SIZE, filled with random numbers
```

Comment:
Please change your code to use /* ... */ comments everywhere, not // comments, for consistency with the rest of the library.